shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -416,93 +507,59 @@ public void setSystemUiMode() {
-                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
-
-    verify(fakeDecorView).setSystemUiVisibility(fakeSetFlags &= ~View.SYSTEM_UI_FLAG_FULLSCREEN);
-    verify(fakeDecorView)
-        .setSystemUiVisibility(fakeSetFlags &= ~View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
-  }
-
-  @TargetApi(30)
-  @Test
-  public void showSystemOverlays() {
-    View fakeDecorView = mock(View.class);
-    Window fakeWindow = mock(Window.class);
-    when(fakeWindow.getDecorView()).thenReturn(fakeDecorView);
-    Activity fakeActivity = mock(Activity.class);
-    when(fakeActivity.getWindow()).thenReturn(fakeWindow);
-    PlatformChannel fakePlatformChannel = mock(PlatformChannel.class);
-    PlatformPlugin platformPlugin = new PlatformPlugin(fakeActivity, fakePlatformChannel);
+    // WindowInsetsControllerCompat windowInsetsControllerCompat = new
+    // WindowInsetsControllerCompat(fakeWindow, testView);
     WindowInsetsController fakeWindowInsetsController = mock(WindowInsetsController.class);
     when(fakeWindow.getInsetsController()).thenReturn(fakeWindowInsetsController);
 
-    platformPlugin.mPlatformMessageHandler.showSystemOverlays(
-        new ArrayList<PlatformChannel.SystemUiOverlay>());
-
-    verify(fakeWindowInsetsController)
-        .setSystemBarsBehavior(WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
-    verify(fakeWindowInsetsController).hide(WindowInsetsCompat.Type.systemBars());
-
-    platformPlugin.mPlatformMessageHandler.showSystemOverlays(
-        new ArrayList<PlatformChannel.SystemUiOverlay>(
-            Arrays.asList(
-                PlatformChannel.SystemUiOverlay.TOP_OVERLAYS,
-                PlatformChannel.SystemUiOverlay.BOTTOM_OVERLAYS)));
-
-    verify(fakeWindowInsetsController).show(WindowInsetsCompat.Type.statusBars());
-    verify(fakeWindowInsetsController).show(WindowInsetsCompat.Type.navigationBars());
+    platformPlugin.mPlatformMessageHandler.setSystemUiChangeListener();
+
+    /* All of the attemps below resulted in the call: fakePlatformCannel.systemChromeCanged(true) */
+
+    // (1) Attempt with using listnener directly (I exposed it by adding a getInsetsListener method)
+    WindowInsets.Builder builder = new WindowInsets.Builder();
+    builder.setInsets(WindowInsetsCompat.Type.systemBars(), Insets.of(0, 200, 0, 200));
+    WindowInsets fullScreenInsets = builder.build();
+
+    platformPlugin
+        .getInsetsListener()
+        .onApplyWindowInsets(testView, WindowInsetsCompat.toWindowInsetsCompat(fullScreenInsets));
+
+    // (1.5) Attempt with setting visibility of system bars
+    testView.getWindowInsetsController().show(WindowInsetsCompat.Type.systemBars());
+    verify(fakePlatformChannel).systemChromeChanged(false);
+
+    // (2) Attempt with setting systemUiMode:
+    // platformPlugin.mPlatformMessageHandler.showSystemUiMode(
+    //     PlatformChannel.SystemUiMode.EDGE_TO_EDGE);
+    // verify(fakePlatformChannel).systemChromeChanged(true);
```

Comment:
This looks like how this should be tested, using the exact methods that change the ui, so you can verify that the listener responds to them as expected. 

In this commented out case with edge to edge, I think it should be returning `fakePlatformCannel.systemChromeCanged(true)` because edge to edge is not full screen.